src/dataflow/src/source/postgres.rs

```diff
@@ -104,50 +110,55 @@ impl PostgresSourceReader {
             .await;
 
         // Get all the relevant tables for this publication
-        let publication_tables =
+        let publication_tables = try_recoverable!(
             postgres_util::publication_info(&self.connector.conn, &self.connector.publication)
-                .await?;
+                .await
+        );
 
         // Start a transaction and immediatelly create a replication slot with the USE SNAPSHOT
         // directive. This makes the starting point of the slot and the snapshot of the transaction
         // identical.
-        client
-            .simple_query("BEGIN READ ONLY ISOLATION LEVEL REPEATABLE READ;")
-            .await?;
+        try_recoverable!(
+            client
+                .simple_query("BEGIN READ ONLY ISOLATION LEVEL REPEATABLE READ;")
```

Comment:
I just merged #6898 that makes all `tokio_postgres::Error` errors work with the `?` operator inside a function that returns `ReplicationError`. This will make sure all errors pass through the finer grained "is it recoverable?" logic in the conversion

Reply:
Updated!